File: app/src/main/java/com/steven/moviesearch/loaders/DetailsLoader.java
package com.steven.moviesearch.loaders;

import android.content.AsyncTaskLoader;
import android.content.Context;

import com.steven.moviesearch.EndPoints;
import com.steven.moviesearch.R;
import com.steven.moviesearch.Utils;
import com.steven.moviesearch.models.ResultItem;

import java.io.IOException;

import okhttp3.OkHttpClient;
import retrofit2.Call;
import retrofit2.Retrofit;

public class DetailsLoader extends AsyncTaskLoader<ResultItem> {

	private static final String TAG = DetailsLoader.class.getName();

	private int resultId;
	private Context context;

	public DetailsLoader(Context context, int resultId) {
		super(context);
		this.context = context;
		this.resultId = resultId;
	}

	@Override
	public ResultItem loadInBackground() {
		OkHttpClient okHttpClient = Utils.provideOkHttpClient();
		Retrofit retrofit = Utils.provideRetrofit(okHttpClient);
		EndPoints endPoints = retrofit.create(EndPoints.class);
		Call<ResultItem> resultItemCall =
				endPoints.getMovieDetails(resultId, context.getString(R.string.theMovieDB_api_key));
		try {
			return resultItemCall.execute().body();
		} catch (IOException e) {
			return null;
		}
	}

	@Override
	protected void onStartLoading() {
		forceLoad();
	}
}

File: app/src/main/java/com/steven/moviesearch/loaders/DetailsLoader.kt
package com.steven.moviesearch.loaders

import android.content.Context
import android.support.v4.content.AsyncTaskLoader
import com.steven.moviesearch.EndPoints
import com.steven.moviesearch.R
import com.steven.moviesearch.Utils
import com.steven.moviesearch.models.ResultItem
import retrofit2.Call
import java.io.IOException

class DetailsLoader(context: Context, private val resultId: Int, private val resultMediaType: String) : AsyncTaskLoader<ResultItem>(context) {

	override fun loadInBackground(): ResultItem? {
		val okHttpClient = Utils.provideOkHttpClient()
		val retrofit = Utils.provideRetrofit(okHttpClient)
		val endPoints = retrofit.create(EndPoints::class.java)
		val resultItemCall: Call<ResultItem>
		if (resultMediaType == "movie") {
			resultItemCall = endPoints.getMovieDetails(resultId, context.getString(R.string.theMovieDB_api_key))
		} else {
			resultItemCall = endPoints.getTVDetails(resultId, context.getString(R.string
					.theMovieDB_api_key))
		}
		try {
			return resultItemCall.execute().body()
		} catch (e: IOException) {
			return null
		}

	}

	override fun onStartLoading() {
		forceLoad()
	}

	companion object {
		private val TAG = DetailsLoader::class.java.name
	}
}
